packages/expo-firebase-core/ios/EXFirebaseCore/EXFirebaseCore.m

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020-present 650 Industries. All rights reserved.
+
+#import <UMCore/UMUtilities.h>
+#import <EXFirebaseCore/EXFirebaseCore.h>
+#import <EXFirebaseCore/EXFirebaseCore+FIROptions.h>
+
+#define DEFAULT_APP_NAME_IOS @"__FIRAPP_DEFAULT"
+#define DEFAULT_APP_NAME_UNIVERSAL @"[DEFAULT]"
+
+@interface EXFirebaseCore ()
+
+@end
+
+@implementation EXFirebaseCore {
+  NSString* _appName;
+  FIROptions* _appOptions;
+}
+
+UM_REGISTER_MODULE();
+
++ (const NSString *)exportedModuleName
+{
+  return @"ExpoFirebaseCore";
+}
+
++ (const NSArray<Protocol *> *)exportedInterfaces
+{
+  return @[@protocol(UMFirebaseCoreInterface)];
+}
+
++ (NSString*) toUniversalAppName:(NSString*)name
+{
+  return [name isEqualToString:DEFAULT_APP_NAME_IOS]
+    ? DEFAULT_APP_NAME_UNIVERSAL
+    : name;
+}
+
++ (NSString*) fromUniversalAppName:(NSString*)name
+{
+  return [name isEqualToString:DEFAULT_APP_NAME_UNIVERSAL]
+    ? DEFAULT_APP_NAME_IOS
+    : name;
+}
+
+- (nonnull instancetype) init
+{
+  if (self = [super init]) {
+    _appName = DEFAULT_APP_NAME_IOS;
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"GoogleService-Info" ofType:@"plist"];
+    if (path && ![FIRApp defaultApp]) {
+      [FIRApp configure];
+    }
+    FIRApp* defaultApp = [FIRApp defaultApp];
+    if (defaultApp) {
+      _appName = defaultApp.name;
+      _appOptions = defaultApp.options;
+    }
+  }
+  return self;
+}
+
+- (nonnull instancetype) initWithAppName:(nonnull NSString*)name options:(nullable FIROptions*)options
+{
+  if (self = [super init]) {
+    _appName = name;
+    _appOptions = options;
+    
+    // Initialize the firebase app. This will delete/create/update the app
+    // if it has changed, and leaves the app untouched when the config
```

Comment:
That's a good question, and one I cannot answer. I specifically did this to ensure that no data "gets lost" when for instance repeatedly refreshing your app. And to provide the best possible performance, as Firebase doesn't need to tear-down, restart any internal services. This might become a thing when more services like Firestore are introduced that persist data to an internal database.